src/App.tsx

```diff
@@ -17,18 +43,18 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter setFilter={setFilter} />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading ? <Loader /> : (
+                <TodoList todos={filteredTodos} selectedTodoId={selectedTodo?.id} onTodoSelect={handleTodoSelect} />
+              )}
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {selectedTodo && <TodoModal selectedTodo={selectedTodo} onResetTodo={resetTodo} />}
```

Comment:
```suggestion
      {!!selectedTodo && <TodoModal selectedTodo={selectedTodo} onResetTodo={resetTodo} />}
```
in rendering with && for best rendering control use !! to return a boolean value